### ISSUE TITLE
[BUG] large amount of SQS queues cause missing metrics

### ISSUE THREAD

**Issue by OliverKlette85** [nerdswords/yet-another-cloudwatch-exporter#489]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

We have around 9,5 k SQS queues in the eu-west-1 region of one of our prod accounts, but the YACE exporter only provides metrics for around 5 K of them.

![image](https://user-images.githubusercontent.com/5149367/144458709-34bfdfbb-823d-4a75-8f3f-11ebbe1743a1.png)

![image](https://user-images.githubusercontent.com/5149367/144458986-21be4141-a6bf-46b6-93b2-f86307899bc9.png)

I already tried to run several YACE instances in parallel:

- split by SQS metrics 
- split by searchTags

Both didn't improve the situation. I also requested quota increase of AWS quotas for GetMetricData (1000 per second) and ListMetrics (100 per second) requests and according to AWS monitoring we are far away from reaching it. 

In the YACE debug log I couldn't find any entries which explain the missing metrics. 

### Expected Behavior

The exporter should provide metrics of all SQS queue (it worked with official Cloudwatch exporter) 

### Steps To Reproduce

config:

```
extraArgs:
  scraping-interval: 120
  debug: true

config: |-
  discovery:
    exportedTagsOnMetrics:
      sqs:
        - dh_app
        - dh_country
        - dh_env
        - dh_platform
        - dh_region
        - dh_squad
        - dh_tribe
    jobs:
    - type: sqs
      regions:
        - eu-west-1
      delay: 600
      period: 120
      length: 120
      awsDimensions:
       - QueueName
      metrics:
        - name: ApproximateAgeOfOldestMessage
          statistics:
          - Average
 ```

**Comment by thomaspeitz** [comment]
Wow thats a hugh amount of SQS queues 🎉

We need more debugging logs to find the error here.
Will try to add the debugging in the next seven days.

Alternative provide me A SEPARATE AWS account with 6k SQS already created to debug this:

Cross account sharing via:
arn:aws:iam::838758336246:user/debug-yace-489

Don't forget to add the permissions for this user as well:
```
"tag:GetResources",
"cloudwatch:GetMetricData",
"cloudwatch:GetMetricStatistics",
"cloudwatch:ListMetrics"
```

Will debug it in the next 7 days.

**Comment by OliverKlette85** [comment]
Hi Thomas, thanks for your quick reaction.

I created a role with desired permissions and a trust policy for your user in our stg account:

`arn:aws:iam::487596255802:role/yace_debug`

This account has actually over 12 k SQS queues in the eu-west-1 region :)

Please ping if you need anything from my side.

**Comment by thomaspeitz** [comment]
```
{"level":"info","msg":"Couldn't describe resources for region eu-west-1: AccessDeniedException: User: arn:aws:sts::487596255802:assumed-role/yace_debug/1638822531663485000 is not authorized to perform: tag:GetResources because no identity-based policy allows the tag:GetResources action\n\tstatus code: 400, request id: f6e26e95-0a6d-4632-b2a6-052425ceeeff\n","time":"2021-12-06T21:28:52+01:00"}
```

Could you double check on your side is everything is configured correctly?
Seems I can switch succesfully into the role but missing the permissions:

```
Don't forget to add the permissions for this user as well:

"tag:GetResources",
"cloudwatch:GetMetricData",
"cloudwatch:GetMetricStatistics",
"cloudwatch:ListMetrics"
```

**Comment by OliverKlette85** [comment]
Yeah I set these rights. Maybe the issue was caused by the condition I set on policy level. I moved it to the trust relationship. Could you please test again?

**Comment by thomaspeitz** [comment]
Okay, I see the issue. Now I have something to debug with. Had two small things in mind which were not the issue.

Need to dig deeper.

Greetings, Thomas :)

**Comment by OliverKlette85** [comment]
Thanks for the efforts! Please let me know if you need something from my side.

Grüße aus Berlin :)

**Comment by thomaspeitz** [comment]
Grüße zurück aus Aachen (aktuell).

FYI: Was not able to put much time into it (and did not find anything yet) and will only work on this at the end of this week again due to private stuff. - Would be nice if you keep the role active so I can debug it further. - Still not understanding what happens their. Was expecting pagination bugs which does not seem to be the problem.

**Comment by OliverKlette85** [comment]
Thanks for the update. I will keep the role active.

**Comment by OliverKlette85** [comment]
Hi @thomaspeitz did you manage to have another look?

**Comment by endyrocket** [comment]
Hi! We're facing the same issue, we have around 350 queues and some of them are entirely ignored. Reverting back to old cloudwatch exporter fixes the issue. We're using version: v0.28.0-alpha

**Comment by eusokolov** [comment]
Hi, we are having the same issue

**Comment by thomaspeitz** [comment]
Sorry was doing vacation. Back again.

@OliverKlette85 if you have the IAM still configured I will take again a look on this topic this week.

**Comment by OliverKlette85** [comment]
Yes it is still active.

**Comment by thomaspeitz** [comment]
Thats gold worth to know "Reverting back to old cloudwatch exporter fixes the issue. We're using version: v0.28.0-alpha" @endyrocket - Thanks! - Makes it easier to debug.

Awesome thank you @OliverKlette85 will work on that probably Thuersday / Friday.

Currently my active work on the project is cut to 2h a week due to no revenue generation from the project. Will try my best to fix this but it is top of the list (at least) to get fixed.

**Comment by nickyfoster** [comment]
@mmanjos I can confirm that adding tags to SQS queue solves this issue.

I had the same problem with my queues not being visible in exported metrics. Turned out those queues had 0 tags on them. After adding arbitrary tag I was able to query metric.
